src/main/java/com/onfleet/demo/homework/util/ObjectGenerator.java

```diff
@@ -90,10 +91,10 @@ public Watermarks(final double latitudeLowMark,
                       final double latitudeHiMark,
                       final double longitudeLowMark,
                       final double longitudeHiMark) {
-      this.latitudeLowMark = latitudeLowMark;
-      this.latitudeHiMark = latitudeHiMark;
-      this.longitudeLowMark = longitudeLowMark;
-      this.longitudeHiMark = longitudeHiMark;
+      this.latitudeLowMark = latitudeLowMark * Watermarks.precisionMultiple;
+      this.latitudeHiMark = latitudeHiMark * Watermarks.precisionMultiple;
+      this.longitudeLowMark = longitudeLowMark * Watermarks.precisionMultiple;
+      this.longitudeHiMark = longitudeHiMark * Watermarks.precisionMultiple;
```

Comment:
slight enhancement to auto-generating locations: precomputed lat/long watermarks at high resolution